cmd/sink/client.go

```diff
@@ -0,0 +1,20 @@
+package main
+
+import (
+	"fmt"
+
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+)
+
+func GetKubernetesClient() (*dynamic.DynamicClient, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("Error retrieving in-cluster k8s client config: %s", err)
+	}
+	client, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("Error instantiating k8s from host %s: %s", config.Host, err)
+	}
+	return client, nil
+}
```

Comment:
We have a very similar code in the API Server in here.

Just noticing it because:
* We may want to unify those
* You are using [dynamic.DynamicClient](https://github.com/kubernetes/client-go/blob/master/dynamic/simple.go#L38) while in the other code I used [kubernetes.ClientSet](https://github.com/kubernetes/client-go/blob/master/kubernetes/clientset.go#L139). We need to see the differences and see if one is better than the other

Reply:
I noticed that as well and I initially attempted to unify them before I realized they were two separate clients. The issue for the sink is that `kubernetes.ClientSet` doesn't provide methods for operating on custom resource, which `dynamic.DynamicClient` does. I'll look at them again, but if `dynamic.DynamicClient` doesn't provide something similar to what you're able to do with the discovery interface, then it might not be possible

Reply:
ok, thank you for the thorough answer, you don't need to worry for the time being to unify it in this PR, but as this isn't obvious I'd appreciate a comment stating that Dynamic client is needed for operating on custom resources

Reply:
Unfortunately, `dynamic.DynamicClient` does not have any knowledge about the resource(s) that you are trying to consume and instead lets you specify a `GroupVersionResource` that you create. This works for the sink because we construct this from an actual k8s object, where the api server receives these values from the user and may not exist. As it stands right now, I'm not sure that it would be possible to merge these two

Reply:
I'll add a doc comment to the function

Reply:
Thank you!